app/models/locker_application.rb

```diff
@@ -58,9 +58,10 @@ def available_lockers_in_area_and_size(building:)
     context.order(:location)
   end
 
-  def self.search(uid:)
-    return where(complete: true) if uid.blank?
-
-    where(complete: true).joins(:user).where("users.uid = '#{uid}'")
+  def self.search(uid:, archived:)
```

Comment:
Right now this method is doing an awful lot of database lookups - is there a way we can streamline this? Maybe something like 

```ruby
def self.search(uid:, archived:)
  # Raise error if archived isn't already a boolean, log and rescue from error and cast archived to a boolean
  return where(complete: true).where(archived: archived) if uid.blank?

  where(archived: archived).where(complete: true).joins(:user).where("users.uid = '#{uid}'")
end
```

You could maybe use this gem to help verify number of database lookups? https://github.com/civiccc/db-query-matchers

Reply:
One complication is we can't use `where(archived: archived)` when archived is true unless we want to change the functionality from hiding archived or showing all results. I also implemented it so archiving was only possible for applications on the assignment page, this code is used for search for applications on the all applications page. Maybe the best course is to separate these into different methods?

Reply:
Refactored to prevent extra database lookups.